dotrix_pbr/src/shaders/light.inc.wgsl

```diff
@@ -70,104 +76,210 @@ fn calculate_point(
         + light.attenuation.z * (light_distance * light_distance)
     );
 
-    return color + (diffuse * light.color.rgb * attenuation);
+    var out: LightCalcOutput;
+    out.light_direction = light_direction;
+    out.radiance = light.color.rgb * attenuation;
+    return out;
 }
 
-
 fn calculate_simple(
     light: SimpleLight,
-    color: vec3<f32>,
     position: vec3<f32>,
     normal: vec3<f32>,
-) -> vec3<f32> {
+) -> LightCalcOutput {
     let light_direction: vec3<f32> = normalize(light.position.xyz - position.xyz);
-    let diffuse: f32 = max(0.0, dot(normal, light_direction));
 
-    return color + diffuse * light.color.rgb;
+    var out: LightCalcOutput;
+    out.light_direction = light_direction;
+    out.radiance = light.color.rgb;
+    return out;
 }
 
 
 fn calculate_spot(
     light: SpotLight,
-    color: vec3<f32>,
     position: vec3<f32>,
     normal: vec3<f32>,
-) -> vec3<f32> {
+) -> LightCalcOutput {
     let light_direction: vec3<f32> = normalize(light.position.xyz - position.xyz);
     let theta: f32 = dot(light_direction, normalize(-light.direction.xyz));
 
     let epsilon: f32 = light.cut_off - light.outer_cut_off;
     let intensity: f32 = clamp((theta - light.outer_cut_off) / epsilon, 0.0, 1.0);
 
-    let diffuse: f32 = max(0.0, dot(normal, light_direction));
-    return color + (diffuse * light.color.xyz) * intensity;
+    var out: LightCalcOutput;
+    out.light_direction = light_direction;
+    out.radiance = light.color.rgb * intensity;
+    return out;
+}
+
+fn DistributionGGX(N: vec3<f32>, H: vec3<f32>, roughness: f32) -> f32
+{
+    let a: f32 = roughness*roughness;
+    let a2: f32 = a*a;
+    let NdotH: f32 = max(dot(N, H), 0.0);
+    let NdotH2: f32 = NdotH*NdotH;
+
+    let num: f32 = a2;
+    var denom: f32 = (NdotH2 * (a2 - 1.0) + 1.0);
+    denom = PI * denom * denom;
+
+    return num / denom;
+}
+
+fn GeometrySchlickGGX(NdotV: f32, roughness: f32) -> f32
+{
+    let r: f32 = (roughness + 1.0);
+    let k: f32 = (r*r) / 8.0;
+
+    let num: f32   = NdotV;
+    let denom: f32 = NdotV * (1.0 - k) + k;
+
+    return num / denom;
+}
+fn GeometrySmith(N: vec3<f32>, V: vec3<f32>, L: vec3<f32>, roughness: f32) -> f32
+{
+    let NdotV: f32 = max(dot(N, V), 0.0);
+    let NdotL: f32 = max(dot(N, L), 0.0);
+    let ggx2: f32  = GeometrySchlickGGX(NdotV, roughness);
+    let ggx1: f32  = GeometrySchlickGGX(NdotL, roughness);
+
+    return ggx1 * ggx2;
 }
 
+// Calulates the amount of light that refects (specular) and that which scatters (diffuse)
+fn fresnelSchlick(cosTheta: f32, F0: vec3<f32>) -> vec3<f32>
```

Comment:
could you please bring all new functions names to a snake_case?

Reply:
Now in snake_case, I tired to add their full names too when possible